internal/cmd/headless_repl/src/processes/invoke_go.rs

```diff
@@ -0,0 +1,130 @@
+use crate::invoke_go::OutputError::InvalidMethod;
+use crate::processes::lsp_invoke::add_headers;
+use crate::processes::process_completion::process_completions_response;
+use crate::LSPSuggestionHelper::LSPSuggestionHelper;
+use crate::{CommandHint, MyHelper};
+use lsp_types::{
+    DidChangeTextDocumentParams, TextDocumentContentChangeEvent, Url,
+    VersionedTextDocumentIdentifier,
+};
+use regex::Regex;
+use rustyline::Helper;
+use serde_json::Value;
+use std::collections::HashSet;
+use std::env::consts::OS;
+use std::fmt::format;
+use std::hash::Hash;
+use std::io::{stdin, Read};
+use std::process::{Child, ChildStdout, Command, Stdio};
+use std::ptr::write;
+use std::str;
+use std::str::from_utf8;
+use std::string::String;
+use std::sync::atomic::AtomicUsize;
+use std::sync::mpsc::Sender;
+use std::sync::{Arc, Mutex, RwLock};
+use tower_lsp::jsonrpc;
+use tower_lsp::jsonrpc::RequestBuilder;
+
+pub fn start_go() -> Child {
+    let mut child = Command::new("./main")
+        .stdin(Stdio::piped())
+        .stdout(Stdio::piped())
+        .spawn()
+        .expect("failure to execute");
+    child
+}
+
+#[derive(Debug)]
+pub enum OutputError {
+    InvalidMethod,
+}
+
+impl From<serde_json::Error> for OutputError {
+    fn from(_: serde_json::Error) -> Self {
+        InvalidMethod
+    }
+}
+
+pub fn form_output(request_type: &str, text: &str) -> Result<String, OutputError> {
+    match request_type {
+        "Service.DidOutput" => {
+            let mut cleaned = text.to_string().replace("\"", "\\\"");
+            let mut param = (r#"[{"input": "#).to_string();
+            let mut other_side = format!(r#""{}""#, cleaned);
+            other_side.push_str("}]");
+            param.push_str(other_side.as_str());
+            let paramm: Value =
+                serde_json::from_str(param.as_str()).expect("failure going to value");
+
+            let req: RequestBuilder = jsonrpc::Request::build("Service.DidOutput").params(paramm);
+            let a = serde_json::to_value(req.finish())?;
+            let res = serde_json::to_string(&a).unwrap();
+
+            // println!("{}", res);
+            Ok(res)
+        }
+        _ => Err(InvalidMethod),
+    }
+}
+
+pub fn read_json_rpc(child_stdout: ChildStdout, mut storage: Arc<RwLock<HashSet<CommandHint>>>) {
+    let re = Regex::new(r"Content-Length: ").unwrap();
+    let num = Regex::new(r"\d").unwrap();
+    let mut buf: Vec<u8> = vec![];
+    let mut num_buf: Vec<u8> = vec![];
+    let mut x = 0;
+    let mut y = 0;
+    //indicate when to start and stop capturing numbers in the content length
+    let mut num_cap = false;
+    let mut read_exact = (false, 0);
+    for i in child_stdout.bytes() {
+        let val = i.unwrap();
```

Comment:
This will panic if there are IO errors, better to return a `Result` from this function and use the `?` operator to propagate it up.

You can use https://docs.rs/anyhow/1.0.61/anyhow/ or `Box<dyn std::error::error + Send + Sync>` as the error type if you do not want to mess with specific ones.

Reply:
@Westonside looks like you marked the error handling feedback as resolved, but the diff doesn't show the change.

You still need to have functions where you're using `unwrap()` or `expect()` return a `anyhow::Result`, then replace the `unwrap()` and `expect()` with `?`.